### PR TITLE
Pull in basic kcov support

### PR DESCRIPTION
**Pull request by cb22**
This brings in `kcov` from #878.

To use it locally: run with `COV=1 zig build test`

On CI:
* See the Summary page for the Core pipeline, it'll have both a diff to main of coverage, and absolute coverage, as well as HTML reports for download.
* It'll give inline annotations on PRs about lines that aren't covered.

### CODE REVIEW

File: build.zig [hunk 1]
```diff
@@ -187,13 +187,28 @@ pub fn build(b: *std.build.Builder) void {
...
         const unit_tests = b.addTest("src/unit_tests.zig");
+        unit_tests.addPackage(vsr_package);
         unit_tests.setTarget(target);
         unit_tests.setBuildMode(mode);
         unit_tests.addOptions("vsr_options", options);
         unit_tests.step.dependOn(&tb_client_header_generate.step);
         unit_tests.setFilter(test_filter);
         link_tracer_backend(unit_tests, tracer_backend, target);
 
+        // Turn on test coverage if COV env var is not blank.
```
Comment (eatonphil) [inline build.zig:195]: This looks familiar. :D

Comment (eatonphil) [reply build.zig:195]: @cb22 could you also pull in my updates to HACKING.md that talk about how to use this?

File: build.zig [hunk 2]
```diff
@@ -190,13 +190,28 @@ pub fn build(b: *std.build.Builder) void {
...
+        if (std.process.getEnvVarOwned(b.allocator, "COV")) |coverage| {
+            b.allocator.free(coverage);
+            unit_tests.setExecCmd(&[_]?[]const u8{
+                "kcov",
+                "--clean",
+                "--include-path=src/",
+                "kcov-output",
```
Comment (eatonphil) [inline build.zig:202]: Should we add this directory to the .gitignore as well?

File: build.zig (hunk 1 above)
Comment (cb22) [reply build.zig:195]: Done!

File: build.zig (hunk 2 above)
Comment (cb22) [reply build.zig:202]: Yep, it is there

**Review by eatonphil — APPROVED** [review]

**Pull request merged**